apps/air-discount-scheme/backend/src/app/modules/nationalRegistry/nationalRegistry.service.ts

```diff
@@ -204,19 +204,24 @@ export class NationalRegistryService {
         return testUser
       }
     }
-
     const cacheKey = this.getCacheKey(nationalId, 'user')
     const cacheValue = await this.cacheManager.get(cacheKey)
     if (cacheValue) {
       return cacheValue.user
     }
-
     const response: {
       data: [NationalRegistryGeneralLookupResponse]
     } = await this.httpService
       .get(`${this.baseUrl}/general-lookup?ssn=${nationalId}`)
       .toPromise()
 
+    if (response && response.data[0].error) {
+      this.logger.error(
+        `National Registry general lookup failed for User<${nationalId}> due to: ${response.data[0].error}`,
+      )
+      return null
+    }
```

Comment:
More swallowing but this is more understandable, since it is used in an N+1 lookup in the backend. Do you want the chance for partial results and hide any errors from the user, or do we want to show all or nothing with a clear error in the UI.

Generally I would return "missing" national ids as null or 404, but throw other errors.

Reply:
My line of thought is to solve the error boundaries of dev/staging while test data is not sufficient, to me it is better to have partial results, silent error and usable site. 

This shouldn't be a problem on production since all individual SSNs exist - however every error would be logged and if no users are returned then the controller would throw error.

Reply:
Yeah, we had a similar issue in the delegation system and followed the same line of thought.

This is something that GraphQL can actually help with. Instead of having one resolver for everything, it's possible to split it up, so if one resolution fails, you might still get partial data while still having the partial error sent to the client/UI. Anyways...